src/NuGet.Core/NuGet.Protocol/HttpSource/HttpSourceAuthenticationHandler.cs

```diff
@@ -95,12 +96,34 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
                 if (response.StatusCode == HttpStatusCode.Unauthorized ||
                     (configuration.PromptOn403 && response.StatusCode == HttpStatusCode.Forbidden))
                 {
+                    IList<Stopwatch> stopwatches = null;
+
+                    if (request.Properties.TryGetValue(HttpRetryHandler.StopwatchPropertyName, out var value))
+                    {
+                        stopwatches = value as IList<Stopwatch>;
+                        if (stopwatches != null)
+                        {
+                            foreach (var stopwatch in stopwatches)
+                            {
+                                stopwatch.Stop();
+                            }
+                        }
+                    }
+
                     promptCredentials = await AcquireCredentialsAsync(
                         response.StatusCode,
                         beforeLockVersion,
                         configuration.Logger,
                         cancellationToken);
 
+                    if (stopwatches != null)
+                    {
+                        foreach (var stopwatch in stopwatches)
```

Comment:
Should the header stopwatch really be restarted?

Reply:
we haven't yet got the header for the request that will get returned in `HttpRetryHandler`, so I say yes.